static/js/typeahead_helper.js

```diff
@@ -34,15 +34,18 @@ export function highlight_with_escaping_and_regex(regex, item) {
     // to know the case of the content we're replacing (you can't just use a bolded
     // version of 'query')
 
-    const pieces = item.split(regex);
+    const pieces = item.split(regex).filter(Boolean);
     let result = "";
 
-    for (const piece of pieces) {
-        if (regex.test(piece)) {
-            result += "<strong>" + Handlebars.Utils.escapeExpression(piece) + "</strong>";
-        } else {
-            result += Handlebars.Utils.escapeExpression(piece);
-        }
+    // We highlight only the prefix of the word if it matches the regex
+    if (regex.test(pieces[0])) {
```

Comment:
Why do you test only the first piece here now?

Reply:
The first piece is essentially the prefix of the item (each of which is word) so only it is tested as mid word matches are irrelevant. For example, if the search term is "the" it'll match the prefix "The" in "**The**ta" but not the midword "the" in "O**the**llo"